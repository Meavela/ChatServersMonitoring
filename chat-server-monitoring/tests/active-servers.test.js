import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ActiveServers from '../src/Components/active-servers';

// Mock the component to take the fake server url
jest.mock('../src/Components/active-servers');

test('Display active servers', () => {
    const {container} = render(<ActiveServers />)
    expect(container.textContent).toMatch('http://127.0.0.1:7419/unittesthttp://127.0.0.1:6592/unitest2')
});

test('Not change active servers', () => {
    const component = renderer.create(
        <ActiveServers />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();  
});