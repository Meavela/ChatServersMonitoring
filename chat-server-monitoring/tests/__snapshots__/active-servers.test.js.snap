// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display active servers 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        http://127.0.0.1:2345
      </span>
    </div>
  </div>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root servers-items MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
      >
        http://127.0.0.1:5432
      </span>
    </div>
  </div>
</ul>
`;
